src/game_api/script/lua_backend.cpp

```diff
@@ -177,6 +178,51 @@ CustomMovableBehavior* LuaBackend::make_custom_movable_behavior(std::string_view
     return custom_behavior.get();
 }
 
+sol::object LuaBackend::get_user_data(Entity& entity)
+{
+    if (user_datas.contains(entity.uid))
+    {
+        return user_datas[entity.uid];
+    }
+    return sol::nil;
+}
+sol::object LuaBackend::get_user_data(uint32_t uid)
+{
+    if (user_datas.contains(uid))
+    {
+        return user_datas[uid];
+    }
+    return sol::nil;
+}
+void LuaBackend::set_user_data(Entity& entity, sol::object user_data)
+{
+    if (!user_datas.contains(entity.uid))
+    {
+        entity.set_on_dtor(
+            [this, &entity](Entity*)
+            {
+                /*auto state = State::get().ptr_local();
```

Comment:
Leftover testing code or still needed to finish up?

Reply:
I tried doing it in the dtor first, which technically works but is stupid if the player doesn't have userdata set, only the carried item. You'd need to figure out in reverse if the item is being carried by a player.

Reply:
Unless of course you hooked the player dtor always, not just when it has userdata

Reply:
Actually things you are carrying might still get destroyed before the player, the other way is much safer.